### DISCUSSION TITLE
AuthUser.firebaseUser delay

### DISCUSSION THREAD

**Discussion by kevinnammour** [gladly-team/next-firebase-auth#632] [Q&A] [answered]
Why does the AuthUser.firebaseUser takes time to be set? At first, it was null for like 2 seconds and then it is set to the current user. Why is that?

**Comment by kmjennison** [comment, marked as answer]
The `firebaseUser` property is the user from the Firebase JS SDK, which takes time to initialize on the client. It will be null before the Firebase JS SDK has initialized.

**Reply by kevinnammour** [reply]
@kmjennison thank you. So if I made a firestore function that runs in a useEffect is a Page component, that might result in an error because the client SDK is not initialized yet? If yes, how can I prevent this?

**Reply by kmjennison** [reply]
Correct. You can wait for `AuthUser.clientInitialized` to be true (or, equivalently, for `AuthUser.firebaseUser` to be defined). You can use the `clientInitialized` property as a dependency in your hook to re-run the effect when that changes.
